### PR TITLE
fix: ensure websocket is flushed

### LINKED ISSUE #6345: eth_subscribe newHeads doesn't appear to work?
### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (3e12d88 2023-11-17T00:33:21.440760000Z)

### What command(s) is the bug in?

anvil -b 1

### Operating System

macOS (Apple Silicon)

### Describe the bug

It appears the latest version of anvil is not correctly handling new block subscriptions.

On the latest version of Anvil `anvil -b 1`, this request comes back with nothing.

```json
{
    "jsonrpc": "2.0",
    "id": 0,
    "method": "eth_subscribe",
    "params": ["newHeads"]
}
```

I discovered this after updating from my old version a few months ago; all of our CI started failing.

I verified this is an issue by installing the previous commit we were using:

```shell
foundryup --commit 7b291d43778be31620a9b45b19a68339f7286f08
```

Then running 

```shell
anvil -b 1
```

New block subscription works as expected.

### PR DESCRIPTION
**Pull request by mattsse**
closes #6345

regression introduced via tungseinite bump in https://github.com/foundry-rs/foundry/pull/6321

ref https://github.com/snapview/tokio-tungstenite/pull/284

TLDR: websocket sink no longer always handles poll_flush

### CODE REVIEW

**Review by Evalir — APPROVED** [review]

**Comment by mattsse** [comment]
no error, it just no longer always flushed the messages to the wire due to some API changes
